packages/statusbar-extension/src/index.ts

```diff
@@ -355,6 +357,29 @@ const modeSwitch: JupyterFrontEndPlugin<void> = {
     shell.modeChanged.connect((_, mode) => {
       modeSwitch.value = mode === 'single-document';
     });
+
+    if (settingRegistry) {
+      const loadSettings = settingRegistry.load(STATUSBAR_PLUGIN_ID);
+      const updateSettings = (settings: ISettingRegistry.ISettings): void => {
+        const startWithSimpleMode = settings.get('startWithSimpleMode')
+          .composite as boolean;
+        if (startWithSimpleMode) {
+          shell.mode = 'single-document';
+        }
+      };
+
+      Promise.all([loadSettings, app.restored])
+        .then(([settings]) => {
+          updateSettings(settings);
+          settings.changed.connect(settings => {
+            updateSettings(settings);
+          });
```

Comment:
This is a special settings that acts once. So do we want to react to a change? What I mean is, if the user is in multi documents mode and turn the setting on, it will be switched to single document. But that may not be the intent.

Reply:
agree. I have changed that setting and the visible mode was toggled. As the name is `startWithSimpleMode` it should only be applied on start. I will remove that update block.